src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs

```diff
@@ -884,7 +884,11 @@ private LLVMValueRef OutputCodeForGetGenericDictionary(LLVMBuilderRef builder, L
             ReadyToRunGenericHelperNode node, GenericLookupResult lookup, LLVMValueRef dictionary)
         {
             // Find the generic dictionary slot
-            int dictionarySlot = factory.GenericDictionaryLayout(node.DictionaryOwner).GetSlotForEntry(lookup);
+            if (!factory.GenericDictionaryLayout(node.DictionaryOwner)
+                    .TryGetSlotForEntry(lookup, out int dictionarySlot))
+            {
+                throw new Exception($"Failed to find slot entry in generic dictionary layout ${node.DictionaryOwner.GetDisplayName()}, entry {lookup}");
+            }
```

Comment:
I don't think so as this is the code to create the dictionary lookup (e.g from a type at runtime), its not called by the jit during compilation, @MichalStrehovsky what do you think?

Reply:
If I understand this correctly, this code is the equivalent of the assembly we emit in `ReadyToRunGenericHelperNode` - I would expect this to generate null lookup instead of throwing.

It might only be reachable when running with `--scan` but not `-O`. It's certainly a corner case. Adding a comment to the exception with a link to dotnet/runtime#82591 might be enough.

Reply:
Thanks, I stand corrected.